pkg/comp-functions/functions/common/mailgun_alerting.go

```diff
@@ -0,0 +1,154 @@
+package common
+
+import (
+	"context"
+	"strconv"
+
+	xkube "github.com/crossplane-contrib/provider-kubernetes/apis/object/v1alpha1"
+	alertmanagerv1alpha1 "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1alpha1"
+	vshnv1 "github.com/vshn/appcat/v4/apis/vshn/v1"
+	runtime "github.com/vshn/appcat/v4/pkg/comp-functions/runtime"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/utils/pointer"
+	controllerruntime "sigs.k8s.io/controller-runtime"
+)
+
+func MailgunAlerting(ctx context.Context, iof *runtime.Runtime) runtime.Result {
+	var name, email, instanceNamespace string
+	log := controllerruntime.LoggerFrom(ctx)
+	log.Info("Starting mailgun-alerting function")
+
+	x, _ := iof.Observed.GetCompositeAlmostGeneric(ctx)
+
+	switch x := x.(type) {
```

Comment:
can't we use the `default` case here, if it's the same with both services we have right now?